core/test/Store.js

```diff
@@ -0,0 +1,66 @@
+const { didContractThrow } = require("../../common/SolidityTestUtils.js");
+
+const ERC20MintableData = require("openzeppelin-solidity/build/contracts/ERC20Mintable.json");
+const truffleAssert = require("truffle-assertions");
+const truffleContract = require("truffle-contract");
+const ERC20Mintable = truffleContract(ERC20MintableData);
+ERC20Mintable.setProvider(web3.currentProvider);
+const Store = artifacts.require("Store");
+
+contract("Store", function(accounts) {
+  // A deployed instance of the Store contract, ready for testing.
+  let store;
+
+  const owner = accounts[0];
+  const derivative = accounts[1];
+  const erc20TokenOwner = accounts[2];
+
+  const identifier = web3.utils.utf8ToHex("id");
+
+  //Add test final fee for test identifier
+
+  beforeEach(async function() {
+    store = await Store.new();
+  });
+
+  it("Compute fees", async function() {
```

Comment:
I lot of the checks below are pretty isolated from one another. WDYT about separating as many as possible into distinct test cases? We've been bad about this in the past, but it would be nice to make our tests more readable :). Even one line test cases could totally make sense!

For example, this could be its own test case:
```
    // Change fee to 20%
    newFee = { value: web3.utils.toWei("0.2", "ether") };
    await store.setFixedOracleFeePerSecond(newFee, { from: owner });

    // Run time tests again
    fees = await store.computeRegularFee(100, 101, pfc, identifier);
    assert.equal(fees.regularFee.toString(), web3.utils.toWei("0.4", "ether"));

    fees = await store.computeRegularFee(100, 110, pfc, identifier);
    assert.equal(fees.regularFee.toString(), web3.utils.toWei("4", "ether"));
```

This could also be its own test case:
```
    // Disallow endTime < startTime.
    assert(await didContractThrow(store.computeRegularFee(2, 1, 10, identifier)));
```